### ISSUE TITLE
Log rotation creates confusing, empty log files.

### ISSUE THREAD

**Issue by funkyfuture** [eXist-db/exist#1662]
### What is the problem

eXist-db creates daily, new log files while still writing into the open file which results in a confusing log directory structure. (users are assuming no logs have been written at all.)

### What did you expect
Contents of a log should be written to a log file that matches the messages' times, empty log files shouldn't persist at all.

### Describe how to reproduce or add a test

We're constantly running several eXist-db instances controlled by `systemd`. No changes to the logging configuration have been applied. They all show the same pattern in their log directory:

```
-rw-r--r-- 1 existdb existdb      0 Dez 19 06:25 exist.log
-rw-r--r-- 1 existdb existdb      0 Dez  5 06:45 exist.log-20171205
-rw-r--r-- 1 existdb existdb 642617 Dez 10 03:03 exist.log-20171206
-rw-r--r-- 1 existdb existdb      0 Dez  7 06:46 exist.log-20171207
-rw-r--r-- 1 existdb existdb      0 Dez  8 06:45 exist.log-20171208
-rw-r--r-- 1 existdb existdb 555421 Dez 13 03:03 exist.log-20171209
-rw-r--r-- 1 existdb existdb      0 Dez 10 06:46 exist.log-20171210
-rw-r--r-- 1 existdb existdb      0 Dez 11 06:46 exist.log-20171211
-rw-r--r-- 1 existdb existdb 465825 Dez 15 15:00 exist.log-20171212
-rw-r--r-- 1 existdb existdb      0 Dez 13 06:47 exist.log-20171213
-rw-r--r-- 1 existdb existdb      0 Dez 14 06:48 exist.log-20171214
-rw-r--r-- 1 existdb existdb 719479 Dez 19 15:15 exist.log-20171215
-rw-r--r-- 1 existdb existdb      0 Dez 16 06:26 exist.log-20171216
-rw-r--r-- 1 existdb existdb      0 Dez 17 06:25 exist.log-20171217
-rw-r--r-- 1 existdb existdb      0 Dez 18 06:25 exist.log-20171218
```

e.g. the file `exist.log-20171206` contains messages from December 6th, 2017 ~15:30 till December 10th of the same year when the host was restarted.

### Context information
Please always add the following information
- eXist-db 3.5.0 /  12aea86 
- OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
- Ubuntu 16.04.3
- 64 bit

**Comment by dizzzz** [comment]
Looks like a log4j2 issue / logging configuration issue.

**Comment by duncdrum** [comment]
I'll just drop this in here, if anything comes out of this that will help addressing [this related issue](eXist-db/documentation/#12)

**Comment by dizzzz** [comment]
404?

**Comment by dizzzz** [comment]
Interesting..... the log files should have been gzipped ....

**Comment by dizzzz** [comment]
Actually, exist does not rotate on a daily basis but size based. This does not yet explain why empty files are generated though. I highly suspect a bug in log4j2 ; Version 2.10 is there, but I don't see any evidence in the changelog.

**Comment by duncdrum** [comment]
@dizzzz apologies for bad link, fixed now

**Comment by dizzzz** [comment]
hint for 0 byte files: https://issues.apache.org/jira/browse/LOG4J2-1501 

`Add a createOnDemand="true|false" attribute to FileAppender. The default is "false" for backward compatibility (which we can revisit of course).`

**Comment by funkyfuture** [comment]
yep, but this is still open: https://issues.apache.org/jira/browse/LOG4J2-1632

some more findings that only help to increase the confusion:

- the files `log4j2.xml` are not `diff`erent than right now in the repo here
- the `exist` log explicitly mentions this file by its full path when an instance is starting (and doesn't mention `log4j` furthermore)
- Jetty's logs have the identical config if i understand correctly, yet they behave different:
  - a new log file is created each day
  - they aren't gzipped either

can i change some configuration to produce debug ouput that may help?

**Comment by duncdrum** [comment]
still unresolved it seems

**Comment by line-o** [comment]
I am quite confident that this is no loner an issue and propose to close as completed.
